Feature auto configuration

manuelplazaspalacio opened the pull request:
Close #1504

manuelplazaspalacio committed:
preparing pull an credentials for autoconfigure feature

manuelplazaspalacio committed:
Solving compilation bugs

manuelplazaspalacio committed:
Checking if credentials are null when checking if they are demo credentials.

manuelplazaspalacio committed:
Getting organisation by phone from api

manuelplazaspalacio committed:
Change getInternationalizedCode to getInternationalizedName

manuelplazaspalacio committed:
Downloading organisation unit by phone and saving it.

manuelplazaspalacio committed:
Add autoconfigured credentials

manuelplazaspalacio committed:
Merge branch 'bug-fix_cnm_variant_compilation' into feature-auto_configuration

# Conflicts:
#	app/src/cnm/java/org/eyeseetea/malariacare/domain/usecase/LoginUseCase.java
#	app/src/cnm/java/org/eyeseetea/malariacare/services/strategies/PushServiceStrategy.java
#	app/src/main/java/org/eyeseetea/malariacare/ProgressActivity.java
#	app/src/main/java/org/eyeseetea/malariacare/data/authentication/AuthenticationManager.java

manuelplazaspalacio committed:
Asking for permissions before start auto configure and if can't autoconfigure init app with demo user.

manuelplazaspalacio committed:
Add auto-configure button in settings activity.

manuelplazaspalacio committed:
Merge branch 'development' into feature-auto_configuration

# Conflicts:
#	app/src/main/java/org/eyeseetea/malariacare/network/ServerAPIController.java

xurxodev commented on app/src/cnm/java/org/eyeseetea/malariacare/strategies/SplashActivityStrategy.java:
```diff
@@ -1,16 +1,209 @@
+                .setMessage(R.string.error_auto_configuration)
+                .setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int arg1) {
+                        continueWithDemoUser();
```
when autoconfiguration fail, the app should not continue with demo user, you must do nothing, you will be in no-login mode

xurxodev reviewed (changes requested):
@manuelplazaspalacio I have requested you some changes. please, can you review it?

manuelplazaspalacio committed:
If user autoconfiguration fails, continue in no-login mode.

manuelplazaspalacio committed:
Merge branch 'development' into feature-auto_configuration

manuelplazaspalacio commented:
Changes done @xurxodev

xurxodev reviewed (changes requested):
@manuelplazaspalacio I have found some problems:
- when on pull process occurred either error the autoconfiguration message is shown but this message should be shown only for autoconfiguration error. I think we should create an AutoconfigurationException in domain for CNM and for avoid realize many changes we can change message parameter in onError usecase callback to Throwable then in other variants get the message and execute the same code and on CNM if Throwable is AutoconfigurationException show the correct dialog.
- if autoconfiguration is completed in push process when ban orgunit is realized an exception occurred and push fail
- When autoconfiguration fail, this is the final step the pull process, then has no sense execute pull again. 
- When autoconfiguration is complete, if I click on autoconfiguration in settings we should call to server again for get orgunit

manuelplazaspalacio committed:
Solving bugs:

-show only  autoconfiguration error when there is an autoconfiguration error.
-login with fixed credntial when autoconfigure for push.
-not  execute pull if autoconfigure fails
-remove organisationUnit when clicking on setting autoconfigure.
-solve bugs in other variants.

manuelplazaspalacio commented:
Solved bugs @xurxodev

xurxodev committed:
retrieve hardcoded credentials from AuthenticationManager

xurxodev reviewed (approved):
thanks @manuelplazaspalacio
